15314f1ddb882700dc9fabc5ca961943/update/sys_script_include_b630a09ddb9020d01206fd641d9619df.xml

```diff
@@ -718,20 +718,355 @@ IllumioUtils.prototype = {
+					returnDict["status"] = true;
+				} else if (response_status == 406) {
+					response = JSON.parse(response.getBody())[0];
+					gs.error("Error while creating label dimension " + labelDimensionData["key"] + " on Illumio PCE for PCE configuration " + pceConfig.getValue("name") + ". " + response["message"] + "");
+					returnDict["status"] = false;
+				} else {
+					gs.error("Error while creating label dimension "+ labelDimensionData["key"] + " on Illumio PCE for PCE configuration " + pceConfig.getValue("name") + ", it will be retried with the next scheduled script execution. Status code - " + response_status + ". Response - " + JSON.stringify(response));
+					returnDict["status"] = false;
+				}
+			}			
+		} catch (exc) {
+			gs.error("Exception while creating label dimension on Illumio PCE. " + exc);
+		}
+		return returnDict;
+	},
+	
+	/**
+	 * Update a label dimension on Illumio PCE
+	 * @returns {Boolean} containing status
+	 */
+	updateLabelDimension: function(labelDimensionData) {
+		var status = false;
+		try {
+			var pceConfig = new GlideRecord("x_illu2_illumio_illumio_pce_conf");
+			if (pceConfig.get(labelDimensionData["illumio_pce_configuration"])) {
+				if (!(pceConfig.mid_server.status == MID_SERVER_UP_STATUS && pceConfig.mid_server.validated == 'true')) {
+					gs.error("The configured MID Server with configuration record " + pceConfig.getUniqueValue() + " is down.");
+					status = false;
+					return status;
+				}
+				
+				var stringParams = {
+					"url": pceConfig.pce_url + "",
+					"href": labelDimensionData["href"]
+				};
+				var pce_authorization = gs.base64Encode(pceConfig.username.getDecryptedValue() + ":" + pceConfig.secret_key.getDecryptedValue());
+				var headers = {
+					"Authorization": "Basic " + pce_authorization
+				};
+				var body = {
+					"display_name": labelDimensionData["display_name_singular"],
+					"display_info": {
+						"initial": labelDimensionData["label_type_initial"],
+						"display_name_plural": labelDimensionData["display_name_plural"]
+					},
+				};
+				var midServer = pceConfig.mid_server.name + "";
+
+				var response = this.makeRestCall("IllumioFetchLabelDimensions", "UpdateLabelDimension", stringParams, headers, body, midServer);
+				var response_status = response.getStatusCode();
+
+				if (response_status >= 204 && response_status < 300) {
+					gs.debug("Label Dimension " + labelDimensionData["key"] + " updated successfully on Illumio PCE for PCE configuration " + pceConfig.getValue("name") + ".");
+					status = true;
+				} else {
+					gs.error("Error while updating label dimension " + labelDimensionData["key"] + " on Illumio PCE for PCE configuration " + pceConfig.getValue("name") + ". Status code - " + response_status + ". Response - " + JSON.stringify(response));
+					status = false;
+				}
+			}
+		} catch (exc) {
+			gs.error("Exception while updating label dimension on Illumio PCE. " + exc);
+			return false;
```

Comment:
No need for this return statement as `status` defaults to false and we return it on the next line.